### ISSUE TITLE
Validation accuracy stuck at 0.6 in my CNN model

### ISSUE THREAD

**Issue by requiem-of-souls** [tensorflow/tensorflow#44024]
I've been creating a CNN to map facial emotions to certain emojis, but after certain epochs I cant seem to get the validation accuracy past the 0.61 mark, while the training accuracy keeps increasing smoothly as required. I can recognize this as an overfitting problem since the validation loss also dips and then keeps increasing, but I cant seem to find methods like regularization or dropouts showing any change in the validation accuracy. So I'm wondering if something is wrong with my model itself, like it being having too many parameters or such? Here is the model:

model = tf.keras.Sequential()

model.add(Conv2D(32, kernel_size = (3,3), input_shape = (48,48,1)))
model.add(Activation('relu'))
model.add(Conv2D(64, kernel_size=(3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Conv2D(128, kernel_size=(3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Conv2D(128, kernel_size=(3, 3)))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(1024))
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(7))
model.add(Activation('softmax'))

Data consists of 48x48 pixel images which has 28701 training instances and 7178 validation instances

**Comment by PrattJena** [comment]
Can you please provide a google colab gist along with the dataset link?

**Comment by ravikyram** [comment]
@requiem-of-souls 

Please, fill [issue template](https://github.com/tensorflow/tensorflow/issues/new/choose).Please, let us know which TF version you are using?

Request you to share colab link or simple standalone code with supporting files to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by requiem-of-souls** [comment]
@ravikyram I'm using TF 2.3.0 in Colab. here is the gist of the model: 

https://gist.github.com/requiem-of-souls/ecdccf36c8cace79bc83e32fd3dbddd8#file-trainer-ipynb 

@PrattJena the dataset is the FER2013 dataset from Kaggle:
https://www.kaggle.com/c/challenges-in-representation-learning-facial-expression-recognition-challenge/data

**Comment by ravikyram** [comment]
@requiem-of-souls 

I am not able to see see the gist from the link you have shared. Please, help me with the reproducible code. Thanks!

**Comment by requiem-of-souls** [comment]
sorry for that, here's the colab file itself https://colab.research.google.com/drive/1dcTbpQZSzl8e9VGAsMnSZYpAo6wqnkMQ?usp=sharing

**Comment by ravikyram** [comment]
@requiem-of-souls 

I tried in colab and i am facing the error (`Found 0 images belonging to 0 classes.`). Can you please provide the data under train and test folder.Please, find the [gist](https://colab.research.google.com/gist/ravikyram/c07a09789197150e65e708627996222b/untitled456.ipynb) for your reference.Thanks!

**Comment by requiem-of-souls** [comment]
@ravikyram try importing this dataset, this includes the actual images rather than csv files:

https://www.kaggle.com/msambare/fer2013

**Comment by PrattJena** [comment]
I havent tried running the code fully yet but my views after seeing your code are you are applying too much of data. Everytime you apply dropout it loses us some data. Sure it makes training faster. But you have applied dropouts in 4 places. And if you actually go to count them at the end you are effectively only training on 6000 images which is way less than even the test data. So i guess the reason for low accuracy is  lot of dropout. I will test my theory and get back to you.
